components/motor/gpio/motor_encoder.go

```diff
@@ -188,44 +186,43 @@ func (m *EncodedMotor) rpmMonitor(ctx context.Context, goalRPM, goalPos float64)
 			return
 		case <-timer.C:
 		}
-		m.stateMu.Lock()
-		if !m.state.regulated {
-			m.stateMu.Unlock()
-			continue
-		}
-		m.stateMu.Unlock()
+
 		pos, err := m.position(ctx, nil)
 		if err != nil {
-			m.logger.Info("error getting encoder position, sleeping then continuing: %w", err)
+			m.logger.CInfo(ctx, "error getting encoder position, sleeping then continuing: %w", err)
 			if !utils.SelectContextOrWait(ctx, 100*time.Millisecond) {
-				m.logger.Info("error sleeping, giving up %w", ctx.Err())
+				m.logger.CInfo(ctx, "error sleeping, giving up %w", ctx.Err())
 				return
 			}
 			continue
 		}
 		now := time.Now().UnixNano()
 
-		lastPowerPct := m.state.lastPowerPct
-
-		if (m.DirectionMoving() == 1 && pos >= goalPos) || (m.DirectionMoving() == -1 && pos <= goalPos) {
+		if (direction == 1 && pos >= goalPos) || (direction == -1 && pos <= goalPos) {
 			// stop motor when at or past goal position
 			if err := m.Stop(ctx, nil); err != nil {
-				m.logger.Error(err)
+				m.logger.CError(ctx, err)
 				return
 			}
 			return
 		}
 
+		m.stateMu.Lock()
+		lastPowerPct := m.state.lastPowerPct
+		m.stateMu.Unlock()
+
 		newPower, err := m.makeAdjustments(ctx, pos, lastPos, goalRPM, goalPos, lastPowerPct, now, lastTime)
 		if err != nil {
-			m.logger.Error(err)
+			m.logger.CError(ctx, err)
 			return
 		}
 
 		lastPos = pos
 		lastTime = now
 
+		m.stateMu.Lock()
 		m.state.lastPowerPct = newPower
```

Comment:
`makeAdjustments` already calls `setPower`, so we do not need to update the state here